### PR TITLE
Trial on sign up emails

### PR DESCRIPTION
**Pull request by tspenov**

### CODE REVIEW

File: lib/sanbase/billing/subscription/sign_up_trial.ex [hunk 1]
```diff
@@ -0,0 +1,128 @@
+defmodule Sanbase.Billing.Subscription.SignUpTrial do
```
Comment (IvanIvanoff) [inline lib/sanbase/billing/subscription/sign_up_trial.ex]: Please add a moduledoc here describing the purpose of this table/module

File: lib/sanbase/billing/subscription/sign_up_trial.ex [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+  require Logger
+
+  alias Sanbase.Auth.User
+  alias Sanbase.Billing.Subscription.PromoTrial
+  alias Sanbase.Repo
+
+  @free_trial_days 14
+  # Free trial plans are Sanbase PRO plans
+  @free_trial_plans [13]
```
Comment (IvanIvanoff) [inline lib/sanbase/billing/subscription/sign_up_trial.ex]: Let's have one separate module that simply holds these numbers so they can be accessed with a name, what do you think?
So here you would be able to just call `@free_trial_plans [PlanMetadata.sanbase_pro()]`

File: lib/sanbase/billing/subscription/sign_up_trial.ex [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+  @day_email_type_map %{
+    3 => :sent_3day_email,
+    11 => :sent_11day_email,
+    14 => :sent_end_trial_email
```
Comment (IvanIvanoff) [inline lib/sanbase/billing/subscription/sign_up_trial.ex]: sen**d** or sen**t** ?

File: lib/sanbase/billing/subscription/sign_up_trial.ex [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    sent_welcome_email: "welcome_email"
+  }
+
+  schema "sign_up_trials" do
+    belongs_to(:user, User)
+
+    field(:sent_welcome_email, :boolean, dafault: false)
+    field(:sent_3day_email, :boolean, dafault: false)
+    field(:sent_11day_email, :boolean, dafault: false)
+    field(:sent_end_trial_email, :boolean, dafault: false)
```
Comment (IvanIvanoff) [inline lib/sanbase/billing/subscription/sign_up_trial.ex]: `has_sent` :?

File: priv/repo/migrations/20200210084320_update_api_and_sandata_product_names.exs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  alias Sanbase.Billing.Product
+
+  def up do
+    Application.ensure_all_started(:tzdata)
+    Application.ensure_all_started(:prometheus_ecto)
+    Sanbase.Prometheus.EctoInstrumenter.setup()
+
+    %{
+      Product.product_api() => "SanAPI by Santiment",
```
Comment (IvanIvanoff) [inline priv/repo/migrations/20200210084320_update_api_and_sandata_product_names.exs]: What's this? Isn't this Neuro?

File: lib/sanbase/billing/subscription/sign_up_trial.ex (hunk 2 above)
Comment (tspenov) [reply lib/sanbase/billing/subscription/sign_up_trial.ex]: Added a few now. Later will create automatic functions based on `product+plan name+interval`

File: lib/sanbase/billing/subscription/sign_up_trial.ex (hunk 4 above)
Comment (tspenov) [reply lib/sanbase/billing/subscription/sign_up_trial.ex]: lets stick to `sent`

File: priv/repo/migrations/20200210084320_update_api_and_sandata_product_names.exs (hunk 5 above)
Comment (tspenov) [reply priv/repo/migrations/20200210084320_update_api_and_sandata_product_names.exs]: Yes,  Neuro becomes `SanAPI`

File: lib/sanbase/billing/subscription/sign_up_trial.ex (hunk 4 above)
Comment (IvanIvanoff) [reply lib/sanbase/billing/subscription/sign_up_trial.ex]: I don't understand what's the meaning here, that's why I ask - does it indicate that it should be sent (thus using `sent` - default `true`) or it has been sent (thus uisng `has_sent` - default `false`)

Comment (tspenov) [reply lib/sanbase/billing/subscription/sign_up_trial.ex]: hmm, I think both mean the same - it is past time so it indicates whether the email has been sent - which in both cases defaults to false and is ticked when is sent.

File: lib/sanbase/billing/subscription/sign_up_trial.ex (hunk 3 above)
Comment (tspenov) [reply lib/sanbase/billing/subscription/sign_up_trial.ex]: email type could is better fit for `send` - but I use the same fields as the schema so I can toggle them dynamically without using one more map for `send` -> `sent`

**Pull request merged**